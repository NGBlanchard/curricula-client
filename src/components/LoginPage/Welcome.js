import React from 'react';

export default class Welcome extends React.Component {
  render() {
    return (
      <div className="welcome-container">
        <h1 className="welcome-header">A Community for Teachers</h1>
        <p className="landing">Curricula is a tool for teachers and instructors who wish to exchange ideas for courses, lessons, and curriculum outlines.</p>
        <p className="directions">Click register above or login below.</p> 
      </div>
    )
  }
}