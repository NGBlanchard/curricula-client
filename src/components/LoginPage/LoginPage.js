import React, { Component } from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'
import { Section } from '../../components/Utils/Utils'
import LoginNav from '../Nav/LoginNav'
import Welcome from './Welcome'

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  onLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
  }

  render() {
    return (
      <>
      <LoginNav />
      <Welcome />
      <Section className='LoginPage'>
      <div className="success-alert">
          {<p className='success'>Your registration was a success :)</p>}
        </div>
        <h2>Login</h2>
        <LoginForm
          onLoginSuccess={this.onLoginSuccess}
        />
      </Section>
      </>
    )
  }
}
