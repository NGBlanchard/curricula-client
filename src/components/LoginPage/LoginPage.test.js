import LoginPage from './LoginPage'
import React from 'react'
import { BrowserRouter as Router } from "react-router-dom";

import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() });

describe('<LoginPage />', () => {
  it('Renders without crashing', () => {
    mount(
      <Router>
        <LoginPage/>
      </Router>
    )
  })
})