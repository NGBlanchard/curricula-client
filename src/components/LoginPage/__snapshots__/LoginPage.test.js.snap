// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders Login Page by default 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LoginPage
      history={
        Object {
          "push": [Function],
        }
      }
      location={Object {}}
    >
      <LoginNav>
        <nav
          className="Nav"
        >
          <h1>
            <Link
              replace={false}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Curricula
              </a>
            </Link>
          </h1>
          <div
            className="Nav__not-logged-in"
          >
            <Link
              replace={false}
              to="/register"
            >
              <a
                href="/register"
                onClick={[Function]}
              >
                Register
              </a>
            </Link>
            <Bull>
              <span
                className="Bull"
              >
                 â€¢ 
              </span>
            </Bull>
            <Link
              replace={false}
              to="/login"
            >
              <a
                href="/login"
                onClick={[Function]}
              >
                Log in
              </a>
            </Link>
          </div>
        </nav>
      </LoginNav>
      <Welcome>
        <div
          className="welcome-container"
        >
          <h1
            className="welcome-header"
          >
            A Community for Teachers
          </h1>
          <p
            className="landing"
          >
            Curricula is a tool for teachers and instructors who wish to exchange ideas for courses, lessons, and curriculum outlines.
          </p>
          <p
            className="directions"
          >
            Click register above or login below.
          </p>
        </div>
      </Welcome>
      <Section
        className="LoginPage"
      >
        <section
          className="Section LoginPage"
        >
          <div
            className="success-alert"
          >
            <p
              className="success"
            >
              Your registration was a success :)
            </p>
          </div>
          <h2>
            Login
          </h2>
          <LoginForm
            onLoginSuccess={[Function]}
          >
            <form
              className="LoginForm"
              onSubmit={[Function]}
            >
              <div
                role="alert"
              />
              <div
                className="user_name"
              >
                <label
                  htmlFor="LoginForm__user_name"
                >
                  User name
                </label>
                <Input
                  id="LoginForm__user_name"
                  name="user_name"
                >
                  <input
                    className="Input "
                    id="LoginForm__user_name"
                    name="user_name"
                  />
                </Input>
              </div>
              <div
                className="password"
              >
                <label
                  htmlFor="LoginForm__password"
                >
                  Password
                </label>
                <Input
                  id="LoginForm__password"
                  name="password"
                  type="password"
                >
                  <input
                    className="Input "
                    id="LoginForm__password"
                    name="password"
                    type="password"
                  />
                </Input>
              </div>
              <Button
                type="submit"
              >
                <button
                  className="Button "
                  type="submit"
                >
                  Login
                </button>
              </Button>
            </form>
          </LoginForm>
        </section>
      </Section>
    </LoginPage>
  </Router>
</BrowserRouter>
`;
